### PR TITLE
python3Packages.squarify: init at 0.4.3

### PR DESCRIPTION
**Pull request by veehaitch**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/squarify/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "squarify";
+  version = "0.4.3";
+
+  src = fetchFromGitHub {
+    owner = "laserson";
+    repo = pname;
+    rev = "v${version}";
```
Comment (Trundle) [inline pkgs/development/python-modules/squarify/default.nix:14]:
```suggestion
    rev = "refs/tags/v${version}";
```

**Comment by Trundle** [comment]
Also seems to have a dependency to `matplotlib` (not tracked in upstream's `setup.py` though)

**Review by Trundle — APPROVED** [review]

**Comment by Trundle** [comment]
Result of `nixpkgs-review pr 219899` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.squarify</li>
    <li>python310Packages.squarify.dist</li>
    <li>python311Packages.squarify</li>
    <li>python311Packages.squarify.dist</li>
  </ul>
</details>

**Comment by veehaitch** [comment]
> Also seems to have a dependency to `matplotlib` (not tracked in upstream's `setup.py` though)

Created a PR upstream: https://github.com/laserson/squarify/pull/37

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/219899#issuecomment-1461557994)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-219899/results/python311Packages.squarify/lib/python3.10/site-packages]$ python
Python 3.10.10 (main, Feb  7 2023, 12:19:31) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import squarify
>>> values = [500, 433, 78, 25, 25, 7]
>>> values.sort(reverse=True)
>>> x = 0.
>>> y = 0.
>>> width = 700.
>>> height = 433.
>>> values = squarify.normalize_sizes(values, width, height)
>>> values
[141900.7490636704, 122886.04868913858, 22136.516853932586, 7095.0374531835205, 7095.0374531835205, 1986.6104868913858]
```

Looks good to me.

**Pull request merged**